v2/pubsub-to-redis/src/main/java/com/google/cloud/teleport/v2/templates/functions/RedisHashIO.java

```diff
@@ -0,0 +1,164 @@
+/*
+ * Copyright (C) 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package com.google.cloud.teleport.v2.templates.functions;
+
+import com.google.auto.value.AutoValue;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import org.apache.beam.sdk.io.redis.RedisConnectionConfiguration;
+import org.apache.beam.sdk.transforms.DoFn;
+import org.apache.beam.sdk.transforms.PTransform;
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PDone;
+import org.apache.beam.vendor.grpc.v1p48p1.com.google.common.base.Preconditions;
+import org.checkerframework.checker.nullness.qual.NonNull;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import redis.clients.jedis.Jedis;
+import redis.clients.jedis.Pipeline;
+import redis.clients.jedis.Transaction;
+
+public abstract class RedisHashIO {
```

Comment:
Why do we need a new client? Can we use https://beam.apache.org/releases/javadoc/current/org/apache/beam/sdk/io/redis/RedisIO.html?

Reply:
We have some use-cases from our customers and we're also talking to Google PM about a possible search use-case and this requires writing to Redis Hashes, so we can create RediSearch indexes on it and search the documents. The RedisIO doesn't have Hash support yet and it's a bigger task to update RedisIO so for the time being I added this Hash support for this template. RedisIO needs lots of changes as Jedis has completely changed now and we can take advantage of all the new features but again this will be a bigger and separate effort.